core/server/master/src/test/java/alluxio/master/metastore/caching/CachingInodeStoreTest.java

```diff
@@ -188,29 +187,30 @@ public void edgeIndexTest() throws Exception {
   @Test
   public void listingCacheManyDirsEviction() throws Exception {
     for (int i = 1; i < CACHE_SIZE * 3; i++) {
-      mStore.writeNewInode(inodeDir(i, TEST_INODE_ID));
+      createInodeDir(i, TEST_INODE_ID);
     }
     assertFalse(mStore.mListingCache.getCachedChildIds(TEST_INODE_ID).isPresent());
   }
 
   @Test
   public void listingCacheBigDirEviction() throws Exception {
-    MutableInodeDirectory bigDir = inodeDir(1, 0);
-    mStore.writeNewInode(bigDir);
+    MutableInodeDirectory bigDir = createInodeDir(1, 0);
     long dirSize = CACHE_SIZE;
-    for (int i = 2; i < 2 + CACHE_SIZE; i++) {
-      mStore.addChild(bigDir.getId(), inodeDir(i, bigDir.getId()));
+    for (int i = 10; i < 10 + dirSize; i++) {
+      mStore.addChild(bigDir.getId(), createInodeDir(i, bigDir.getId()));
```

Comment:
before we were adding inodes as children but not creating them with `mStore.WriteNewInode`. This causes listing to be flaky because edges read from the backing store are only valid if their inodes exists.